extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTableWriter.java

```diff
@@ -1087,150 +1089,118 @@ public void close() {
         }
     }
 
-    static class StringTransfer implements TransferObject<Binary[]> {
-
+    /**
+     * Used as a base class for transfer object types like strings or big integers that need specialized encoding or
+     * codecs.
+     */
+    abstract static class EncodedTransfer<T> implements TransferObject<Binary[]> {
         private final ChunkSource.GetContext context;
-        private ObjectChunk<String, Values> chunk;
+        private ObjectChunk<T, Values> chunk;
         private Binary[] buffer;
         private final ColumnSource<?> columnSource;
 
-        /**
-         * Stores the maximum number of values in a single page
-         */
-        private final int maxValuesPerPage;
-
         /**
          * Stores the maximum size of data in a single page. For strings, we don't know in advance what the length of
          * each entry would be, therefore we need to keep track of both maximum size and number of values.
          */
         private final int targetPageSize;
 
         /**
-         * Total number of chunks buffered so far
+         * Index of next string to be buffered from the string chunk
```

Comment:
This is a generic class now, so `String` references should go away in favor of something generic.

Reply:
Nice catch, thanks!